grails-app/views/occurrence/_map.gsp

```diff
@@ -391,6 +430,29 @@
                 once = false;
             }
         });
+
+        // Copy URL button handler
+        $('#copyWmsParameters').click(function() {
+            var wmsParams = $('#wmsParams');
+            wmsParams.select();
+            document.execCommand('copy');
```

Comment:
document.execCommand is now deprecated and you have to use navigator.clipboard instead which is async, so I asked HAL and he replaced with:

$('#copyWmsParameters').click(function () {
    const wmsParams = $('#wmsParams').val(); // Get the value of the textarea
    const button = $(this);
    const originalText = button.text();

    // Use Clipboard API to copy text
    navigator.clipboard.writeText(wmsParams)
        .then(() => {
            // Show temporary success message
            button.text('<g:message code="map.wms.btn.copied" default="Copied!"/>');
            setTimeout(() => {
                button.text(originalText);
            }, 2000);
        })
        .catch(err => {
            console.error('Failed to copy text: ', err);
            alert('Failed to copy text. Please try again.');
        });
});

Reply:
Done